docs/proposals/device-crd.md

```diff
@@ -0,0 +1,556 @@
+// capabilities and access mechanism via property visitors.
+type DeviceModelSpec struct {
+  // List of device attributes.
+  // Required.
+  DeviceProperties       []DeviceProperty        `json:"properties,omitempty"`
+  // List of property visitors which describe how to access the device attributes.
+  // Required.
+  DevicePropertyVisitors []DevicePropertyVisitor `json:"propertyVisitors,omitempty"`
+}
+
+// DeviceProperty describes an individual device property / attribute like temperature / humidity etc.
+type DeviceProperty struct {
+  // The property name.
+  // Required.
+  Name        string            `json:"name,omitempty"`
+  // The property description.
+  // +optional
+  Description string            `json:"description,omitempty"`
+  // Additional metadata about the device property like datatype, range, min, max etc.
+  // Optional: If no metadata is provided, it is assumed that the property is an empty read-only string.
+  // +optional
+  Metadata    map[string]string `json:"metadata,omitempty"`
+}
+
+// DevicePropertyVisitor describes the specifics of accessing a particular device
+// property. Visitors are intended to be consumed by device mappers which connect to devices
+// and collect data / perform actions on the device. A device may support multiple protocols.
+// We can define multiple visitors per protocol for a device. For e.g If a device supports Bluetooth
+// protocol , the visitors would describe the 128-bit UUIDs to read / write charactersitics of a Bluetooth
+// service exposed by the device.
+type DevicePropertyVisitor struct {
+  // The protocol type to connect to the device and access the property.
+  // Required.
+  Protocol     ProtocolType      `json:"protocol,omitempty"`
+  // A list of visitors describing the properties which can be accessed with the protocol.
+  // Required.
+  Visitors     []PropertyVisitor `json:"visitors,omitempty"`
+}
+
+// ProtocolType describes the protocol type used to communicate with the device instance.
+type ProtocolType string
+
+const (
+	// Bluetooth Low Energy(BLE) Protocol
+	ProtocolTypeBluetoothLE  ProtocolType = "BluetoothLE"
+	// Zigbee Protocol
+	ProtocolTypeZigbee       ProtocolType = "Zigbee"
+	// Modbus RTU Protocol
+	ProtocolTypeModbusRTU    ProtocolType = "ModbusRTU"
+        // Modbus TCP Protocol
+	ProtocolTypeModbusTCP    ProtocolType = "ModbusTCP"
+	// BACnet Protocol
+	ProtocolTypeBACnet       ProtocolType = "BACnet"
+	// OPC UA Protocol
+	ProtocolTypeOPCUA        ProtocolType = "OPCUA"
+)
+
+// PropertyVisitor contains details like which property can be accessed and the
+// access mechanisms are described via key-value pairs in the metadata.
+type PropertyVisitor struct {
+  // The device property visitor name.
+  // Required.
+  Name         string            `json:"name,omitempty"`
+  // The device property name to be accessed. This should refer to one of the
```

Comment:
device property -> device attributes.

Reply:
as pointed out earlier , we are trying alignment with mozilla web of things.